ipa-core/src/helpers/stream/chunks.rs

```diff
@@ -371,64 +363,41 @@ impl<const N: usize, T> ChunkBuffer<N> for Vec<T> {
 
 /// Stream returned by `process_stream_by_chunks`.
 #[pin_project]
-struct StreamChunkProcessor<St, T, B, K, F, Fut, D, const N: usize>
+struct StreamChunkProcessor<St, T, B, K, F, Fut, const N: usize>
 where
     St: Stream<Item = Result<T, Error>> + Send,
     B: ChunkBuffer<N>,
     F: Fn(usize, B::Chunk) -> Fut,
     Fut: Future<Output = Result<K, Error>>,
-    D: Fn() -> T,
 {
     #[pin]
     stream: St,
     buffer: B,
     process_fn: F,
-    pad_record_fn: Option<D>,
+    terminated: bool,
```

Comment:
If you make `stream: Fuse<St>`, then that will track this state for you and you can call `[is_terminated](https://docs.rs/futures/latest/futures/stream/trait.FusedStream.html#tymethod.is_terminated)` on that.

Reply:
There is a comment about this on lines 396-397:

> We must implement our own fuse (rather than use `.fuse()` on the inner stream) for the case where we terminate early due to an error.

Using the inner stream as a fuse only works if we poll it until it terminates.